2021-04-30 12:35:11,645 INFO     ---------------------------------------------------------------------------------------------
2021-04-30 12:35:11,645 INFO     Geo: cqd
2021-04-30 12:35:11,645 INFO     Data Path: data/NELL-q2b
2021-04-30 12:35:11,646 INFO     #entity: 63361
2021-04-30 12:35:11,646 INFO     #relation: 400
2021-04-30 12:35:11,646 INFO     #max steps: 1000
2021-04-30 12:35:11,646 INFO     Evaluate unoins using: DNF
2021-04-30 12:35:11,646 INFO     loading data
2021-04-30 12:35:23,944 INFO     Training info:
2021-04-30 12:35:23,945 INFO     Validation info:
2021-04-30 12:35:23,945 INFO     3p: 4000
2021-04-30 12:35:23,946 INFO     Test info:
2021-04-30 12:35:23,946 INFO     3p: 4000
2021-04-30 12:35:25,302 INFO     Model Parameter Configuration:
2021-04-30 12:35:25,302 INFO     Parameter embeddings.0.weight: torch.Size([63361, 2000]), require_grad = True
2021-04-30 12:35:25,302 INFO     Parameter embeddings.1.weight: torch.Size([400, 2000]), require_grad = True
2021-04-30 12:35:25,303 INFO     Parameter Number: 127522000
2021-04-30 12:35:27,746 INFO     Loading checkpoint models/nell-q2b...
2021-04-30 12:35:28,386 INFO     tasks = 3p
2021-04-30 12:35:28,386 INFO     init_step = 99999
2021-04-30 12:35:28,386 INFO     batch_size = 1000
2021-04-30 12:35:28,386 INFO     hidden_dim = 1000
2021-04-30 12:35:28,386 INFO     gamma = 12.000000
2021-04-30 12:35:28,386 INFO     Evaluating on Valid Dataset...
  0%|          | 0/4000 [00:00<?, ?it/s]2021-04-30 12:35:28,607 INFO     Evaluating the model... (0/4000)
  0%|          | 1/4000 [00:00<14:36,  4.56it/s]
main.py --do_valid --do_test --data_path data/NELL-q2b -n 1 -b 1000 -d 1000 -lr 0.1 --disable_warmup --max_steps 1000 --cpu_num 0 --geo cqd --valid_steps 20 --tasks 3p --print_on_screen --test_batch_size 1 --optimizer Adagrad --reg_weight 0.05 --log_steps 5 --checkpoint_path models/nell-q2b --cqd d2 --cqd-t-norm prod --cqd-normalize --cqd-sigmoid --cqd-k 64 --no-save --cuda
overwritting args.save_path
logging to models/nell-q2b
99999
Traceback (most recent call last):
  File "main.py", line 508, in <module>
    main(parse_args())
  File "main.py", line 497, in main
    valid_all_metrics = evaluate(model, valid_easy_answers, valid_hard_answers, args, valid_dataloader, query_name_dict, 'Valid', step, writer)
  File "main.py", line 165, in evaluate
    metrics = KGReasoning.test_step(model, tp_answers, fn_answers, args, dataloader, query_name_dict)
  File "/home/pminervi/workspace/KGReasoning/models.py", line 654, in test_step
    _, negative_logit, _, idxs = model(None, negative_sample, None, batch_queries_dict, batch_idxs_dict)
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/pminervi/workspace/KGReasoning/cqd.py", line 356, in forward
    k=self.k, t_norm=t_norm)
  File "/home/pminervi/workspace/KGReasoning/d2.py", line 147, in query_3p
    scoring_function=scoring_function)
  File "/home/pminervi/workspace/KGReasoning/d2.py", line 32, in score_candidates
    atom_scores_2d = scoring_function(s_emb, p_emb, candidates_emb)
  File "/home/pminervi/workspace/KGReasoning/cqd.py", line 333, in scoring_function
    res, _ = self.score_o(lhs_, rel_, rhs_)
  File "/home/pminervi/workspace/KGReasoning/cqd.py", line 114, in score_o
    return score_1 + score_2, factors
RuntimeError: CUDA out of memory. Tried to allocate 992.00 MiB (GPU 0; 11.91 GiB total capacity; 9.89 GiB already allocated; 237.25 MiB free; 10.79 GiB reserved in total by PyTorch)
